plugins/wasmedge_stablediffusion/sd_func.cpp

```diff
@@ -396,28 +501,53 @@ Expect<uint32_t> SDImageToImage::body(
                     sample_method_t(SampleMethod), SampleSteps, Strength, Seed,
                     BatchCount, ControlImage, ControlStrength, StyleRatio,
                     NormalizeInput, InputIdImagesDir.data());
-  // TODO: upscale image
+  // Upscale image
+  if (UpscaleModelPathLen != 0) {
+    MEM_SPAN_CHECK(UpscaleModelSpan, MemInst, char, UpscaleModelPathPtr,
+                   UpscaleModelPathLen,
+                   "Failed when accessing the Upscaler Image memory."sv)
+    std::string UpscaleModelPath(UpscaleModelSpan.begin(),
+                                 UpscaleModelSpan.end());
+    upscalerModel(UpscaleModelPath.data(), UpscaleRepeats, NThreads, Wtype,
+                  BatchCount, Results);
+  }
+  // save results
   int Len;
   unsigned char *Png = stbi_write_png_to_mem(
-      reinterpret_cast<const unsigned char *>(Results->data), 0, Results->width,
+      reinterpret_cast<const unsigned char *>(Results), 0, Results->width,
       Results->height, Results->channel, &Len, nullptr);
-  if (OutputPathLen != 0) {
```

Comment:
Have you verified the Rust SDK will never pass an empty output path? If not, we still need to check the output path is valid.